packages/kbn-alerts/src/hooks/use_get_alerts_permissions/index.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect, useState } from 'react';
+import { RecursiveReadonly } from '@kbn/utility-types';
+
+export interface UseGetUserAlertsPermissionsProps {
+  crud: boolean;
+  read: boolean;
+  loading: boolean;
+}
+
+export const useGetUserAlertsPermissions = (
+  uiCapabilities: RecursiveReadonly<Record<string, any>>,
+  featureId: string
```

Comment:
I think this logic should be different to allow o11y to be able to use it. we need to remember that they can have multiple alerts from different feature IDs like logs, APM, Metrics etc. Make sense to us but not sure if it will be good for them.